### ISSUE TITLE
Add option to auto-disable if no `.eslintrc`

### ISSUE THREAD

**Issue by chrisdothtml** [Microsoft/vscode-eslint#446]
When I open a repo that doesn't use eslint, I get an obtrusive `output` error from this plugin, requiring me to manually disable it every single time.

It'd be really nice to have an option for this plugin to auto-disable itself if no `.eslintrc` is present in the repo (and/or if it's not in the dependencies)

**Comment by dbaeumer** [comment]
This message is only shown once per new workspace. On the next open the output channel shouldn't come to front. You can alternatively disable eslint as a user setting and enable on the workspace you need to.

**Comment by chrisdothtml** [comment]
> This message is only shown once per new workspace

Exactly. So when, for example, vscode restarts for an update, and I had 4 windows open, I'll get the error for each window, which gets annoying quick. This is especially noticeable because I use the vscode insiders build, which has daily updates.

The way [vscode-standardjs](https://github.com/standard/vscode-standardjs) handles this is a simple solution. Setting `standard.usePackageJson` (which is `false` by default) will tell the plugin to only enable itself if `standard` exists in your `package.json`.

I can make a PR if you're willing to accept that as a solution

**Comment by dbaeumer** [comment]
@chrisdothtml in that example the message should be printed to the Output channel, but the channel shouldn't come to front. So it shouldn't be annoying.

Deciding if a valid eslint configuration file exists in a workspace is quite complicated since ESLint itself has many ways where such a configuration can be stored (e.g. .eslintrc.* file in any parent directory chain, a property in the package.json, ...).

This is why I stayed away from implementing this since it will put the ESLint extension into a catch-up game with the ESLint npm module (if they add a new method or a new lookup rule the extension has to provide this as well). 

So what we could look at is adding a command to the output (e.g. via a command://... link) that when clicked disables the extension for this workspace.

**Comment by chrisdothtml** [comment]
> in that example the message should be printed to the Output channel, but the channel shouldn't come to front

It *does* though; at least for me. Not sure if that's because we a different version of vscode or what.

> Deciding if a valid eslint configuration file exists in a workspace is quite complicated

It doesn't check whether **configuration** is present, but rather the **dependency**. This is much more reliable because most projects won't work without it being in the dependencies. The only edge case is if eslint is globally installed, but in that case, the user can just leave this setting off

**Comment by dbaeumer** [comment]
@chrisdothtml thanks for the clarification.

I tested it and on my machine using VS Code 1.23 and ESLint latest the Output channel comes only to front once. However the message is printed everytime.

**Comment by tell-y** [comment]
I tried to set `eslint.workingDirectories: [{ "mode": "auto" }]` option to resolve the topic.
It seems it is correctly working. Is it the correct way? @dbaeumer

**Comment by dbaeumer** [comment]
@tell-y yes and no. In your setup the eslint server is still started. The feature request is to not start the server at all if there is no `.eslintrc` file.

**Comment by ira-gordin-sap** [comment]
What is the current status of this issue?

**Comment by dbaeumer** [comment]
The problem is that ESLint has quite some locations where a eslint configuration can be stored. It is not only `.eslintrc` file in various directories but also `package.json` files. If we only want to check for `.eslintrc.*` files in the root directory then it is easy (see https://github.com/microsoft/vscode-eslint/blob/release/2.1.x/client/src/extension.ts#L838). But this will very likely not work for mono repos which became popular.

**Comment by iRoachie** [comment]
@dbaeumer Outside of checking for multiple locations, do you see any other cases that might block this?

I was also thinking maybe it's an optional toggle, and v1 could be just .eslintrc with the others to follow

**Comment by dbaeumer** [comment]
The major problem is that this put the extension into a catchup with the ESLint library. If they add a new config location or a new way to define the config we need to honor this. This is why in general I stay away from this and such a feature can only go behind a flag and must be off by default.

**Comment by Florian-Schoenherr** [comment]
Just saying, this would align with what prettier has:
![image](https://user-images.githubusercontent.com/65456722/107118041-ae0a2300-687e-11eb-8f0a-07c880118de6.png)

**Comment by Meligy** [comment]
I think I came here for this too.

I get an error message when I open a folder without node_modules directly under it, about a server not existing.

But I think it shouldn't have tried to start this server without a package.json in the first place. Typically, I'd use the eslint inside the repo for this.

For some reason, a global install of eslint does not seem to fix this.
